third_party/move/move-compiler-v2/src/function_checker.rs

```diff
@@ -89,156 +93,175 @@ pub fn check_access_and_use(env: &mut GlobalEnv) {
-                                        env.diag_with_labels(
-                                            Severity::Error,
-                                            &callee_env.get_loc(),
-                                            &msg,
-                                            call_details,
-                                        );
-                                    }
+                                    generic_error(
+                                        env,
+                                        "a script",
+                                        "it is not public",
+                                        sites,
+                                        &callee_env,
+                                    );
                                     false
                                 },
                                 Visibility::Friend => {
                                     if callee_env.module_env.has_friend(&caller_module_id) {
                                         true
                                     } else {
-                                        if caller_is_inline || callee_is_inline {
-                                            let call_details: Vec<_> = sites
-                                                .iter()
-                                                .map(|node_id| {
-                                                    (
-                                                        env.get_node_loc(*node_id),
-                                                        "called here".to_owned(),
-                                                    )
-                                                })
-                                                .collect();
-                                            let msg = format!(
-                                                "`public(friend)` {}function `{}` cannot be called from {}function `{}` because module `{}` is not a `friend` of `{}`",
-                                                if callee_is_inline { "inline " } else { "" },
-                                                callee_env.get_full_name_with_address(),
-                                                caller_name_str,
-                                                if caller_is_inline { "inline " } else { "" },
-                                                caller_module_name.display_full(env),
-                                                callee_env.module_env.get_full_name_str());
-                                            env.diag_with_labels(
-                                                Severity::Error,
-                                                &callee_env.get_loc(),
-                                                &msg,
-                                                call_details,
-                                            );
-                                        }
+                                        let why = format!(
```

Comment:
For the sake of consistency, pull this out into `not_friend_to_module_error()` or some such (mimicking `private_to_module_error()` below)?